### ISSUE TITLE
Questions for pre-release

### ISSUE THREAD

**Issue by mcgregorian1** [ncsuSEAL/Bayesian_LSP#3]
I will be adding more things as I think of them.

These questions are for the test data.
- Why only the snow flag?
- Which QA should be used if not using Landsat data?
- From the documentation for `FitBLSP`, it doesn't seem like the QA column is inherently needed?
- Do the column names need to be the same for someone else’s test?

Other
- What is the exact data that can be used by others? E.g. only Landsat evi? Or Sentinel 2 evi?

FitBLSP
- What is the number that is returned inherently with the function? I get the list with "fitted" and "phenos", but I'm also getting a number that I think is number of iterations to convergence?

**Comment by MrJGao** [comment]
My answers to the questions:

1. Other QA controls had been done when preprocessing the time series. Snow pixels are in the final time series b/c we wanted to see the number of usable observations and many LSP processing algorithms used snow pixels with lower weights. So, although we don't use snow pixels by default, we keep them in the time series and let users control whether they want to use them.
2. It depends on the data source. Basically, QA filters should be used to get all clear observations, i.e., no cloud/cloud shadow, reasonable quality, etc.
3. Do you mean the `test_ts.rda`? I should clean that, it was from my original workflow, for the package, we only need `date`, `evi2`, and `snow`.
4. No, they don't. Users only need to specify the `date_vec` and `vi_vec` columns, and they could assign`weights_vec` if needed.
5. It doesn't rely on any specific data source, the algorithm should fit all vegetation time series that follow the double-logistic function. For example, NDVI, EVI, EVI2 time series from sensors such as Landsat, Sentinel 2, and PlanetScope should all work.
6. The return is a list containing two objects: `fitted` and `phenos`. `fitted` will only be returned if the user sets `ifplot = TRUE` b/c it will generate all possible fitted curves by using all possible combinations of estimated parameters. It takes time and will plot the 95% CI of the fitted curve. `pheno` only contains all estimated phenometrics, it is a data frame. There is no information about the internal fitting process or convergence returned.

**Comment by mcgregorian1** [comment]
Thanks! More questions
- Do we want users to be able to do in-depth customization? E.g. currently the `initValues` argument to `Fit_BLSP` has a default of NULL, with the option for users to supplement their own MCMC starting parameters. If we want to keep this as is, then we will need full documentation of what the starting parameters could be and directions on how to choose them. Otherwise, if we always want the initial pars to be determined by the average model, then we should remove this as a possible argument.

**Comment by MrJGao** [comment]
The `initValues` are just guesses, they should represent a reasonable double logistic curve to start with. Good `initValues` will reduce the computational time needed to converge the model. By default, we fit an average model to estimate the `initValues`, but if the fit failed, we would just use `NULL`, which is the last option we can use. 

So, I suggest keeping the `initValues` as it is, and adding some explanations about how to set it. I think serveral sentences would be enough: 1. the parameters represent a reasonable double-logistic curve for the time series; 2. good `initValues` would reduce the computational cost for the model to coverge; 3. if having no idea, just use `NULL`. 

Thanks!

**Comment by mcgregorian1** [comment]
How sparse can the input time series be? I recommend we figure this out so I can add the documentation in the function help

**Comment by MrJGao** [comment]
Data sparsity is very hard to measure, and data distribution is more important as the ideal case would be that the observations are equally distributed throughout the year. So, I guess users have to try their time series with the BLSP model to see if it can fit well. However, I did some tests, and I found that even if we use data from 1984 to 1994, within which the Landsat data were really sparse, the BLSP model can still fit well. 

I've updated the vignette as well.
